### PR TITLE
Adding utility to create compliant science filenames

### PR DESCRIPTION
**Pull request by ehsteve**
This addresses issue #10 as well as issue #11.

### CODE REVIEW

**Comment by ehsteve** [comment]
Looks like tests are failing only for Python 3.7 because of some issue with the `mock` module which `pytest` uses. Investigating the problem.

File: hermes_core/util/util.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    hermes_{inst}_{mode}_{level}{test}_{descriptor}_{time}_v{version}
+
+    Parameters
+    ----------
+    instrument : `str`
+        The instrument name. Must be one of the following "eea", "nemesis", "merit", "spani"
+    time : `str` (in isot format) or ~astropy.time
+        The time
+    level : str
+        The data level. Must be one of the following
```
Comment (acrager) [inline hermes_core/util/util.py:23]: Need to finish your sentence here.:)

File: hermes_core/util/util.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    """Return a compliant filename root (without extension). The format is defined as
+
+    hermes_{inst}_{mode}_{level}{test}_{descriptor}_{time}_v{version}
+
+    Parameters
+    ----------
+    instrument : `str`
+        The instrument name. Must be one of the following "eea", "nemesis", "merit", "spani"
+    time : `str` (in isot format) or ~astropy.time
+        The time
```
[the remark this answers is not in the document]
Comment (ehsteve) [reply hermes_core/util/util.py:21]: Sorry. What does your comment mean? Not sure I understand.

**Comment by ehsteve** [comment]
Thank you for your review @acrager but you need to be meaner! Hold approval to when testing passes ;)

**Comment by ehsteve** [comment]
I am going to add to the scope of this merge request by also addressing issue #13 which is very related.

File: hermes_core/tests/test_util_util.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+        # wrong instrument name
+        util.create_science_filename("eeb", good_time, level=good_level, version=good_version)
+        util.create_science_filename("fpi", good_time, level=good_level, version=good_version)
+        util.create_science_filename("potato", good_time, level=good_level, version=good_version)
+
+        # bad time string
+        # non-existent time
+        util.create_science_filename(
+            "eeb", "2023-13-04T12:06:21", level=good_level, version=good_version
```
Comment (swkre) [inline hermes_core/tests/test_util_util.py:87]:
```suggestion
            good_instrument, "2023-13-04T12:06:21", level=good_level, version=good_version
```

Comment (swkre) [reply hermes_core/tests/test_util_util.py:87]: Fix on similar 'eeb' starting lines below

**Review by swkre — CHANGES_REQUESTED** [review]

File: hermes_core/util/util.py (hunk 1 above)
Comment (ehsteve) [reply hermes_core/util/util.py]: It has been

File: hermes_core/util/util.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
+"""
+This module provides general utility functions.
+"""
+from astropy.time import Time
+
+__all__ = ["create_science_filename"]
+
+
+def create_science_filename(
+    instrument, time: str, mode="", level="l0", version="", descriptor="", test=False
```
[the remark this answers is not in the document]
Comment (ehsteve) [reply hermes_core/util/util.py]: Done.

File: hermes_core/util/util.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (acrager) [reply hermes_core/util/util.py:25]: I originally thought that "The time" was an unfinished sentence. Just ignore this comment.

**Review by acrager — APPROVED** [review]
All tests passed. Noted a little non-impacting typo in a test utility function name that might just be how Steve C. shortens "output", who knows.

File: hermes_core/tests/test_util_util.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        util.create_science_filename(
+            "eeb",
+            time=12345345,
+            level=good_level,
+            version=good_version,
+            descriptor="blue_green",
+        )
+
+
+def test_parse_science_filename_ouput():
```
[the remark this answers is not in the document]
Comment (ehsteve) [reply hermes_core/tests/test_util_util.py]: Good catch! Thanks. Fixed it and added doc strings as you recommended.

**Pull request merged**